pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -583,6 +584,26 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return *result, err
 	}
 
+	// Sanity check the platform/cloud credentials.
+	validCreds, err := r.validatePlatformCreds(cd, cdLog)
+	if err != nil {
+		cdLog.WithError(err).Error("errored validating platform credentials")
+		return reconcile.Result{}, err
+	}
+	// Make sure the condition is set properly.
+	changed, err := r.setAuthenticationFailure(cd, validCreds, cdLog)
+	if changed || err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// If the platform credentials are no good, do not bother with ClusterProvision objects
+	authCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.AuthenticationFailureClusterDeploymentCondition)
+	if authCondition != nil && authCondition.Status == corev1.ConditionTrue {
+		cdLog.Info("Skipping provision while platform credentials authentication is failing.")
+		// Periodically retry???
```

Comment:
It would be really nice to just requeue every 5 min and be done with it, but there's no reason to think that would help. It probably needs to be done right, watch secrets for changes to a platform secret, requeue the CD if we spot one.

Reply:
We now return an error when credentials are not valid.